### DISCUSSION TITLE
How to build without protobuf

### DISCUSSION THREAD

**Discussion by cvilas** [eclipse-ecal/ecal#1860] [General]
The readme says

> eCAL is message protocol agnostic. You choose the message protocol that fits to your needs like [Google Protobuf](https://developers.google.com/protocol-buffers), [CapnProto](https://capnproto.org/), [Flatbuffers](https://google.github.io/flatbuffers/)...

But, I am unable to build ecal from source without building protobuf or having protobuf on the host. It is actually possible?

**Comment by rex-schilasky** [comment]
Hi, current eCAL stable (5.13) is still using Protobuf for its own internal communication (discovery). So from user perspective you can 'use' eCAL with only Protobuf installed on your host.
Future eCAL version (eCAL6, release planned end of January) will use NanoPb Protobuf library statically linked. So the eCAL runtime has no more Protobuf dependencies finally.

**Reply by rex-schilasky** [reply]
Why do you build eCAL from source?

**Reply by cvilas** [reply]
Thanks @rex-schilasky. Will give eCAL 6 a go when it becomes available.

> Why do you build eCAL from source?

A bit surprised by this question, to be honest. I thought this was standard across the industry. Both at work and for personal projects, I tend to build 'the world' from source, certainly for third-party dependencies of core libraries. Helps me control development environment and deployments across multiple targets and architectures.

**Reply by rex-schilasky** [reply]
Standard across the industry for sure but I assumed you just use it for a personal project.
Did you you succeed to build it finally?
eCAL 6 is worth to upgrade to. It will come with a renewed API as well as significant performance improvements.

**Reply by cvilas** [reply]
Thanks very much. I will give v6.0 a try when it is out.

**Reply by cvilas** [reply]
@rex-schilasky is there a development/pre-release branch where I could try building version 6 already? thanks!

**Reply by rex-schilasky** [reply]
Not yet, you could build the current master branch, this is 'near to' the upcoming v6 release.

**Reply by cvilas** [reply]
Unfortunately master does not build. I guess I will wait for a release. 

```
  Could not find a package configuration file provided by "CMakeFunctions"
  with any of the following names:

    CMakeFunctionsConfig.cmake
    cmakefunctions-config.cmake
```

**Reply by rex-schilasky** [reply]
The documentation is not released yet for eCAL v6, the current documentation fits for v5 only. To build eCAL v6 (master) you need to define a dependency provider.  You can check the github actions how the builds are realized for Ubuntu/Windows and macOS. Finally it's an additional CMake option:

`-DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=cmake/submodule_dependencies.cmake`

What OS you want to build eCAL master for?

**Reply by cvilas** [reply]
Thanks! That worked.

I build on x86-64 (Ubuntu 24.04), Aarch64 (Ubuntu 24.04, Fedora 41, MacOS)
